src/core/sagas/recommendations.ts

```diff
@@ -49,56 +59,110 @@ export namespace Tasks {
+  // tslint:disable-next-line max-line-length
+  export function* fetchPastPurchaseProducts(flux: FluxCapacitor, action: Actions.FetchPastPurchaseProducts, getNavigations: boolean = false) {
     try {
-      const url = `http://${flux.config.customerId}.groupbycloud.com/orders/public/_search`;
-      const response = yield effects.call(fetch, url, Adapter.buildBody(<any>{
-        // slide 18 of Past Purchase Epic?
-        cartType: 'online',
-        // pagination stuff
-        skip: 0,
-        pageSize: 100,
-      }));
-      const result = yield response.json();
-      // TODO: modify data so it's in the right form?
-      yield effects.put(flux.actions.receiveOrderHistory(result.result[0].items));
+      const pastPurchaseSkus: Store.PastPurchases.PastPurchaseProduct[] = yield effects.select(Selectors.pastPurchases);
+      if (pastPurchaseSkus.length > 0) {
+        const request = yield effects.select(Requests.pastPurchaseProducts, getNavigations);
+        const results = yield effects.call(fetchProductsFromSkus, flux, pastPurchaseSkus, request);
+        if (getNavigations) {
+          const navigations = Adapter.pastPurchaseNavigations(
+            yield effects.select(Selectors.config), SearchAdapter.combineNavigations(results));
+          yield effects.put(<any>[
+            flux.actions.receivePastPurchaseAllRecordCount(results.totalRecordCount),
+            flux.actions.receivePastPurchaseRefinements(navigations),
+          ]);
+        } else {
+          yield effects.put(<any>[
+            flux.actions.receivePastPurchasePage(SearchAdapter.extractPage(
+              flux.store.getState(), SearchAdapter.extractRecordCount(results),
+              Selectors.pastPurchasePage,Selectors.pastPurchasePageSize)),
+            flux.actions.receivePastPurchaseCurrentRecordCount(results.totalRecordCount),
+            flux.actions.receivePastPurchaseProducts(SearchAdapter.augmentProducts(results)),
+          ]);
+          flux.saveState(utils.Routes.PAST_PURCHASE);
+        }
+      }
+    } catch (e) {
+      return effects.put(flux.actions.receivePastPurchaseProducts(e));
+    }
+  }
 
-      return [];
+  export function* fetchSaytPastPurchases(flux: FluxCapacitor, { payload }: Actions.FetchSaytPastPurchases) {
+    try {
+      const config: Configuration = yield effects.select(Selectors.config);
+      const pastPurchaseSkus = yield effects.select(Selectors.pastPurchases);
+      if (pastPurchaseSkus.length > 0) {
+        const request = yield effects.select(Requests.autocompleteProducts);
+        const results = yield effects.call(fetchProductsFromSkus, flux, pastPurchaseSkus, {
+          ...request,
+          query: payload
+        });
+        yield effects.put(flux.actions.receiveSaytPastPurchases(SearchAdapter.augmentProducts(results)));
+      } else {
+        yield effects.put(flux.actions.receiveSaytPastPurchases([]));
```

Comment:
so when pastPurchaseSkus is empty, flux still could issue the action FETCH_SAYT_PAST_PURCHASES? Can you eliminate this possibility from the beginning so you don't need to deal with it here?

Reply:
I think this may be the cleanest way, rather than have that logic at every instance of FETCH_SAYT_PAST_PURCHASES being dispatched.

Reply:
but why would FETCH_SAYT_PAST_PURCHASES being dispatched when pastPurchaseSkus is empty? Is it when skus are cleared out? Just trying to understand here.